app/controllers/tool_groups_controller.rb

```diff
@@ -0,0 +1,85 @@
+class ToolGroupsController < ApplicationController
+  before_action :authorize!
+  before_action :convert_hyphen_to_dash, only: [:create, :update]
+
+  def index
+    render json: tool_groups_ordered_by_name, include: params[:include], fields: field_params, status: :ok
+  end
+
+  def create
+    create_tool_group
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def create_tool
+    tool_group_id = params[:tool_group_id].to_i
+    resource_id = params.dig(:data, :relationships, :tool, :data, :id).to_i
+    suggestions_weight = params.dig(:data, :attributes, "suggestions-weight").to_f
+
+    resource_tool_group = ResourceToolGroup.create!(
+      tool_group_id: tool_group_id,
+      resource_id: resource_id,
+      suggestions_weight: suggestions_weight
+    )
+
+    response.headers["Location"] = "tool-groups/#{resource_tool_group.id}"
+    render json: resource_tool_group, status: :created
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def update_tool
+    existing = resource_tool_group
+    existing.update!(
+      resource_id: params[:data][:attributes]["resource-id"],
+      suggestions_weight: params[:data][:attributes]["suggestions-weight"]
+    )
+    render json: existing, status: :accepted
+  end
+
+  def delete_tool
+    resource_tool_group.destroy!
```

Comment:
NoMethodError: undefined method `destroy!' for nil:NilClass

https://app.rollbar.com/a/Cru/fix/item/mobile-content-api/322

<img width="766" alt="Screenshot 2023-08-02 at 2 20 08 PM" src="https://github.com/CruGlobal/mobile-content-api/assets/47818/418c2dc7-a6a6-4b5b-be8c-25baca7302af">

Reply:
this happens when trying to delete a relationship that doesn't exist

Reply:
also, it appears that this logic ignore which tool group the relationship is being deleted from, so I can delete a relationship from a different tool group.

Scenario:
- tool group: 1 relationship id: 1
- tool group: 2 relationship id: 2
- The following delete request should fail: `DELETE /tool-groups/2/tools/1` because relationship 1 doesn't exist for tool-group 2
   - Currently this request will succeed because it doesn't validate that the requested relationship is actually part of the specified tool-group

I haven't verified this, but this same behavior is probably missing from `update_tool` as well because it also uses the same `resource_tool_group` function.

Reply:
should it return a message error or just not explode?

Reply:
fixed